File: 0849-maximize-distance-to-closest-person/0849-maximize-distance-to-closest-person.java
class Solution {
    public int maxDistToClosest(int[] seats) {
        int lastPersonIndex = -1;
        int maxDist = 0;
        for(int i = 0; i <= seats.length - 1; i++){

            if(seats[i] == 1){
                if(lastPersonIndex == -1){ //firstpersonHit
                    maxDist = Math.max(maxDist, i);
                }else{//alreadyfoundAPerson
                    maxDist = Math.max(maxDist,  (i - lastPersonIndex)/2);
                }
                lastPersonIndex = i;
            
            }   
        }
        return maxDist = Math.max(maxDist, (seats.length - lastPersonIndex - 1));
    }
}

File: 0849-maximize-distance-to-closest-person/0849-maximize-distance-to-closest-person.py
class Solution(object):
    def maxDistToClosest(self, seats):
        """
        :type seats: List[int]
        :rtype: int
        """
        lastPersonIndex = -1
        maxDist = 0
        for i in range(len(seats)):
            if seats[i] == 1:
                if lastPersonIndex == -1:
                    maxDist = max(maxDist, i)
                else:
                    maxDist = max(maxDist, (i - lastPersonIndex) //2)
                lastPersonIndex = i
        
        maxDist = max(maxDist, len(seats) - lastPersonIndex - 1)

        return maxDist

        